etl/steps/data/garden/animal_welfare/2023-08-14/number_of_farmed_fish.py

```diff
@@ -0,0 +1,117 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from owid.catalog import Dataset, Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Column names to select and how to rename them.
+COLUMNS = {
+    "country": "country",
+    "year": "year",
+    "estimated_numbers__millions__lower": "n_farmed_fish_low",
+    "estimated_numbers__millions__upper": "n_farmed_fish_high",
+}
+
+# Regions to create aggregates for.
+REGIONS_TO_ADD = [
+    "North America",
+    "South America",
+    "Europe",
+    "European Union (27)",
+    "Africa",
+    "Asia",
+    "Oceania",
+    "Low-income countries",
+    "Upper-middle-income countries",
+    "Lower-middle-income countries",
+    "High-income countries",
+]
+
+
+def add_region_aggregates(tb: Table, ds_regions: Dataset, ds_income_groups: Dataset) -> Table:
+    tb = tb.copy()
```

Comment:
This is something I commonly do. It's not a great solution, but it ensures that, whatever you do to `tb` within the function, will not affect the original `tb`. Sometimes, while debugging or doing some analysis, you may do something like:
```python
tb = Table(...)
test = add_region_aggregates()
...
```
Maybe you now want to compare `tb` and `test`. But if you are not careful, the function `add_region_aggregates` may have modified the original `tb` object. In other words, the function has unintended side effects.
One way to solve that is to be very careful while creating the function, ensuring that there are no side effects. Or another is to simply make a copy of the input object, `tb`, so that whatever happens inside the function stays inside the function. For big tables, this is a very inefficient solution (since it creates a copy of the input table). But for small tables, it's a simple trick.

Reply:
Ah, now I see, thanks. It will depend on the name I assign when I call the function. I assumed it would always be tb

Reply:
If at the beginning you do `tb_new = tb` that may still modify the original `tb`. If instead you make a copy, it doesn't matter how you call the new object. For simplicity, I just call it the same thing.